### DISCUSSION TITLE
Comparison Documentation

### DISCUSSION THREAD

**Discussion by Malix-Labs** [wxt-dev/wxt#493] [Ideas]
Hello folks

It would be very nice to have a comparison documentation with the competition (such as [plasmo](https://github.com/PlasmoHQ/plasmo)), and also state why not using bundler plugins (such as [CRXJS](https://github.com/crxjs/chrome-extension-tools))

It would also be nice to note there your motivation as of why you started developing WXT rather than using the already developed [plasmo](https://github.com/PlasmoHQ/plasmo)
And state what do WXT solve better (and maybe also solve worse _for now_)

I think it would also make more people consider tradeoffs between both, and ultimately make them switch to better tech

## References

- [star history](https://star-history.com/#wxt-dev/wxt&PlasmoHQ/plasmo&crxjs/chrome-extension-tools&Timeline)

**Comment by aklinker1** [comment]
This already exists:
https://wxt.dev/guide/compare.html

I could add comparisons to bundler plugins as well.

**Reply by Malix-Labs** [reply]
Nice, sorry for the overlook

Indeed, bundler plugins would be a nice addition

**Reply by aklinker1** [reply]
No worries! CRXJS is the biggest one, so I'll probably just add that one.

**Reply by Malix-Labs** [reply]
Great news!

As bundler plugins for browser extensions, I know only of CRXJS to be honest

By the way, did you knew both Plasmo and CRXJS before starting WXT?

If yes, it would be nice to also state the considerations that made you create WXT instead of using them, and how does WXT solves those problems better

**Reply by aklinker1** [reply]
Yup, I knew of them, and I think the comparison table shows the differences and features I've added to WXT to make it better, instead of using one of them. But to summarize:

1. Opening the browser automatically during development is a huge time save
2. Compared to CRXJS, it's ESM content script loader makes the trade-off between build speed and respecting the `run_at` option for content scripts - I wanted both
3. Compared to Plasmo, it's vue and svelte support are sketchy, and not as polished as react. Since I use Vue primarily, I needed a better DX
4. I'm a Firefox main, so CRXJS's lack of support for other browsers was a problem
5. New MV3 scripting APIs could provide a better way to reload content scripts, which none of the other tools use to improve dev mode

**Reply by aklinker1** [reply]
I've updated the comparsion page. Let me know if there's any comparisons you think are important that I'm missing.

**Reply by Malix-Labs** [reply]
Saw the update
It's a nice addition and interesting to read!

Are there any technical difference that made you prefer to start WXT from scratch instead of starting with a fork of Plasmo (and perhaps even contributing to upstream)?

**Reply by aklinker1** [reply]
Vite has emerged as the top bundler the last few years, so rather than forking and working with parcel, I chose to start from scratch with Vite.

**Reply by Malix-Labs** [reply]
Very understandable then.
Thanks!

**Reply by ayub-kokabi** [reply]
> This already exists: https://wxt.dev/guide/compare.html
> 
> I could add comparisons to bundler plugins as well.

New compare link:
https://wxt.dev/get-started/compare.html

**Comment by ultrox** [comment]
For anyone looking there is a new page:

https://wxt.dev/guide/resources/compare.html

BTW what I really like about WTX is smaller dependency surface. I work in enterprise env and a lot of things are blocked. Plasmo could not be installed because of sharp (which I was able to override) and node-gyp which finally made me quit.

Wtx went smooth as butter. Thanks for your work

**Reply by Malix-Labs** [reply]
WXT is definitely the current best way dev tool to build browser extension

Also see my thread and answer about that :
https://github.com/crxjs/chrome-extension-tools/discussions/872#discussioncomment-10964530
